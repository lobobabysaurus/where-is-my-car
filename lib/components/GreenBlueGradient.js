import { LinearGradient } from 'expo'
import PropTypes from 'prop-types'
import React from 'react'

const GreenBlueGradient = ({children, style}) => {
  const startColor = '#50C878'
  const endColor = '#00008B'
  const gradient = [startColor, endColor]
  return <LinearGradient style={style} colors={gradient}>
    {children}
  </LinearGradient>
}

GreenBlueGradient.propTypes = {
  children: PropTypes.node,
  style: PropTypes.object,
}

export default GreenBlueGradient
