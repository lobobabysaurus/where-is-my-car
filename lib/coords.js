import store from 'react-native-simple-store'
import { Alert } from 'react-native'

const coordsKey = 'coordinates'

export const saveCoords = async coord => {
  try {
    return await store.save(coordsKey, coord)
  } catch (err) {
    Alert.alert(`Error saving coordinates ${err.message}`)
  }
  return undefined
}

export const getCoords = async () => {
  try {
    return await store.get(coordsKey)
  } catch (err) {
    Alert.alert(`Error getting coordinate: ${err.message}`)
  }
  return undefined
}
