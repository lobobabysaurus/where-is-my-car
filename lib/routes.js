export const MAP = 'Map'
export const HISTORY = 'History'
