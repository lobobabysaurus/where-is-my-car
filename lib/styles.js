import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  button: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center'
  },
  buttonContainer: {
    flexBasis: '10%',
  },
  container: {
    flex: 1,
    flexDirection: 'column',
  },
  header: {
    alignItems: 'flex-end',
    flex: 1,
    flexDirection: 'row',
  },
  headerButton: {
    justifyContent: 'center',
  },
  headerContainer: {
    flexBasis: '10%',
  },
  headerSection: {
    flexBasis: '33%',
  },
  map: {
    flexBasis: '90%',
  },
  text: {
    color: 'white',
    fontSize: 36,
  },
})
