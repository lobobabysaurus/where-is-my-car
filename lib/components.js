import { Location, MapView } from 'expo'
import PropTypes from 'prop-types'
import React from 'react'
import { TouchableHighlight, Text } from 'react-native'

import styles from './styles'

export const MarkButton = ({onPress}) => (
  <TouchableHighlight style={styles.button} onPress={onPress}>
    <Text style={styles.text}>Mark</Text>
  </TouchableHighlight>
)

MarkButton.propTypes = {
  onPress: PropTypes.func.isRequired,
}

export class MarkedMap extends React.Component {

  state = {
    address: undefined,
  }

  componentDidMount = async () => {
    const {coords} = this.props
    this._updateAddress(coords)
  }

  componentDidUpdate = async (oldProps) => {
    const {coords} = this.props
    if (coords && (coords !== oldProps.coords)) {
      this._updateAddress(coords)
    }
  }

  render() {
    const {coords, region} = this.props
    const marker = coords
      ? <MapView.Marker
        coordinate={coords}
        title="Car"
        description={this._getMarkerDescription()} />
      : undefined
    return(
      <MapView style={styles.map} initialRegion={region}>
        {marker}
      </MapView>
    )
  }

  _getMarkerDescription = () => {
    let location = 'Here'
    if (this.state.address) {
      location = `On ${this.state.address.street}`
    }
    return `You Parked ${location}!`
  }

  _updateAddress = async (coords) => {
    if (coords) {
      const addresses = await Location.reverseGeocodeAsync(coords)
      if (addresses.length) {
        const address = addresses[0]
        this.setState({address})
      }
    }
  }
}

MarkedMap.propTypes = {
  coords: PropTypes.shape({
    latitude: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired
  }),
  region: PropTypes.shape({
    latitude: PropTypes.number.isRequired,
    latitudeDelta: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired,
    longitudeDelta: PropTypes.number.isRequired,
  }).isRequired,
}
