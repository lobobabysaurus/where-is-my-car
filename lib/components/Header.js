import PropTypes from 'prop-types'
import React from 'react'
import { TouchableHighlight, Text, View } from 'react-native'

import styles from '../styles'

import GreenBlueGradient from './GreenBlueGradient'

export const HeaderButton = ({onPress, children}) =>
  <TouchableHighlight
    style={styles.headerButton}
    onPress={onPress}>
    <Text style={styles.text}>{children}</Text>
  </TouchableHighlight>

HeaderButton.propTypes = {
  children: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
}

const HeaderSection = ({children}) =>
  <View style={styles.headerSection}>{children}</View>

HeaderSection.propTypes = {
  children: PropTypes.node,
}

const Header = ({leftContent, title, rightContent}) =>
  <GreenBlueGradient style={styles.headerContainer}>
    <View style={styles.header}>
      <HeaderSection>{leftContent}</HeaderSection>
      <HeaderSection><Text style={styles.text}>{title}</Text></HeaderSection>
      <HeaderSection>{rightContent}</HeaderSection>
    </View>
  </GreenBlueGradient>

Header.propTypes = {
  leftContent: PropTypes.node,
  rightContent: PropTypes.node,
  title: PropTypes.string.isRequired,
}

export default Header
