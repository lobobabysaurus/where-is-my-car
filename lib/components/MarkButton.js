import PropTypes from 'prop-types'
import React from 'react'
import { TouchableHighlight, Text } from 'react-native'

import styles from '../styles'

import GreenBlueGradient from './GreenBlueGradient'

const MarkButton = ({onPress}) =>
  <GreenBlueGradient style={styles.buttonContainer}>
    <TouchableHighlight style={styles.button} onPress={onPress}>
      <Text style={styles.text}>Mark</Text>
    </TouchableHighlight>
  </GreenBlueGradient>

MarkButton.propTypes = {
  onPress: PropTypes.func.isRequired,
}

export default MarkButton
