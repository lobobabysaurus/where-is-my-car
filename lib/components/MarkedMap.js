import { Location, MapView } from 'expo'
import PropTypes from 'prop-types'
import React from 'react'

import styles from '../styles'

export default class MarkedMap extends React.Component {

  state = {
    address: undefined,
  }

  componentDidMount = async () => {
    const {coords} = this.props
    this._updateAddress(coords)
  }

  componentDidUpdate = async (oldProps) => {
    const {coords} = this.props
    if (coords && (coords !== oldProps.coords)) {
      this._updateAddress(coords)
    }
  }

  render() {
    const {coords, region} = this.props
    const marker = coords
      ? <MapView.Marker
        coordinate={coords}
        title="Car"
        description={this._getMarkerDescription()} />
      : undefined
    return(
      <MapView style={styles.map} initialRegion={region}>
        {marker}
      </MapView>
    )
  }

  _getMarkerDescription = () => {
    let location = 'Here'
    if (this.state.address) {
      location = `On ${this.state.address.street}`
    }
    return `You Parked ${location}!`
  }

  _updateAddress = async (coords) => {
    if (coords) {
      const addresses = await Location.reverseGeocodeAsync(coords)
      if (addresses.length) {
        const address = addresses[0]
        this.setState({address})
      }
    }
  }
}

MarkedMap.propTypes = {
  coords: PropTypes.shape({
    latitude: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired
  }),
  region: PropTypes.shape({
    latitude: PropTypes.number.isRequired,
    latitudeDelta: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired,
    longitudeDelta: PropTypes.number.isRequired,
  }).isRequired,
}
