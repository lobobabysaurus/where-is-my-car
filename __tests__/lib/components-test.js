import React from 'react'
import renderer from 'react-test-renderer'

import { MarkButton, MarkedMap } from '../../lib/components'

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react');
  const MapView = require.requireActual('react-native-maps');

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children);
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children);
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes;
  MockMarker.propTypes = MapView.Marker.propTypes;
  MockMapView.propTypes = MapView.propTypes;
  MockMapView.Marker = MockMarker;
  MockMapView.Callout = MockCallout;
  return MockMapView;
});

describe('MarkButton', () => {
  test('renders correctly', () => {
    const onPress = () => {}
    const component = renderer.create(<MarkButton onPress={onPress} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})

describe('MarkedMap', () => {
  const region = {
    latitude: 0,
    latitudeDelta: 1,
    longitude: 0,
    longitudeDelta: 1,
  }

  test('renders correctly', () => {
    const component = renderer.create(<MarkedMap region={region} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
