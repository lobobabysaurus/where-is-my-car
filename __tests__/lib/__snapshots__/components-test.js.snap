// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkButton renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "purple",
      "height": "10%",
      "justifyContent": "space-evenly",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 36,
      }
    }
  >
    Mark
  </Text>
</View>
`;

exports[`MarkedMap renders correctly 1`] = `
<MapView
  initialRegion={
    Object {
      "latitude": 0,
      "latitudeDelta": 1,
      "longitude": 0,
      "longitudeDelta": 1,
    }
  }
  style={
    Object {
      "height": "90%",
      "width": "100%",
    }
  }
/>
`;
